pkg/test/executor/docker_entrypoint_test.go

```diff
@@ -0,0 +1,281 @@
+	entrypoint []string
+	cmd        []string
+}
+
+var defaultUserOverwrites = userOverwrites{
+	entrypoint: []string{"/bin/ls"},
+	cmd:        []string{"media"},
+}
+
+func createDockerfile(d dockerfilePermutation) string {
+	ep := "\nENTRYPOINT [\"/bin/echo\"]"
+	cmd := "\nCMD [\"echo\", \"This is from CMD\"]"
+	baseDockerFile := "FROM alpine:latest"
+	if d.entrypoint == true {
+		baseDockerFile += ep
+	}
+	if d.cmd == true {
+		baseDockerFile += cmd
+	}
+	return baseDockerFile
+}
+
+func (suite *DockerEntrypointTestSuite) SetupSuite() {
+	docker.MustHaveDocker(suite.T())
+	ctx := context.Background()
+	suite.testName = strings.ToLower(suite.Suite.T().Name())
+
+	tempDir := suite.T().TempDir()
+	dockerfilePermutations := []struct {
+		entrypoint bool
+		cmd        bool
+	}{
+		{true, true},
+		{true, false},
+		{false, true},
+		{false, false},
+	}
+
+	for _, permutation := range dockerfilePermutations {
+		//create dockerfile
+		dockerFile := createDockerfile(permutation)
+		createUniqueName := func(name string) string {
+			return fmt.Sprintf("%s-%s-%s-%s", suite.testName, name,
+				strconv.FormatBool(permutation.entrypoint),
+				strconv.FormatBool(permutation.cmd))
+		}
+		name := createUniqueName("Dockerfile")
+		filename := filepath.Join(tempDir, name)
+		f, err := os.OpenFile(filename, os.O_CREATE|os.O_WRONLY, util.OS_ALL_RWX)
+		require.NoError(suite.T(), err, "creating temp file")
+		_, err = f.WriteString(dockerFile)
+		require.NoError(suite.T(), err, "writing string to file")
+		data, err := os.ReadFile(filename)
+		require.NoError(suite.T(), err, "reading file")
+		suite.T().Log(string(data))
+
+		f.Close()
+		//build image
+		cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+		require.NoError(suite.T(), err, "Creating client")
+		buildContext := &bytes.Buffer{}
+		err = targzip.Compress(ctx, tempDir, buildContext)
+		require.NoError(suite.T(), err, "Creating build context")
+		tag := createUniqueName("image")
+		buildOptions := types.ImageBuildOptions{
+			Dockerfile: filepath.Join(tempDir, name),
+			Tags:       []string{tag + ":latest"},
+		}
+
+		response, err := cli.ImageBuild(ctx, buildContext, buildOptions)
+		//olgibbons delete after testing:
+		if err != nil {
+			suite.T().Fatal("Error building image: ", err)
+		}
+		output, err := io.ReadAll(response.Body)
+		require.NoError(suite.T(), err, "building image")
+		suite.T().Logf("Image %q built successfully: %s", tag, string(output))
+		defer response.Body.Close()
+	}
+}
+
+func (suite *DockerEntrypointTestSuite) TearDownSuite() {
+	// delete the test docker images here!
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	require.NoError(suite.T(), err)
+
+	imageTags := []string{
+		suite.testName + "-image-true-true",
+		suite.testName + "-image-false-true",
+		suite.testName + "-image-true-false",
+		suite.testName + "-image-false-false",
+	}
```

Comment:
This is a little bit brittle because it relies on:
* The logic of `createUniqueName`
* The `dockerfilePermutations`
I.e. if you change either of those things, this will be broken.

Could you keep a list of all the images created instead? Or could you use something like `suite.T().Cleanup`?